### PR TITLE
Minimal fee amount filter.

### LINKED ISSUE #734: Raise minimal fee up to 1000000
Decline all transactions with lower fee.

### PR DESCRIPTION
**Pull request by oskin1**
Close #734

### CODE REVIEW

File: src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala [hunk 1]
```diff
@@ -54,18 +55,37 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)
...
+        new Exception(
+          s"Minimal fee amount not met. ${minFee.toDouble / den} Ergo required, ${fee.toDouble / den} Ergo given")
+      )
     }
   }
 
+  private def extractFee(tx: ErgoTransaction): Long = {
+    val propositionBytes = settings.chainSettings.monetary.feePropositionBytes
+    ErgoState.boxChanges(Seq(tx))._2
+      .filter(b => java.util.Arrays.equals(b.propositionBytes, propositionBytes))
```
Comment (catena2w) [inline src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala]:
```suggestion
      .filter(_.ergoTree == settings.chainSettings.monetary.feeProposition)
```

File: src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala [hunk 2]
```diff
@@ -37,24 +37,37 @@ class ErgoMemPoolSpec extends FlatSpec
...
+
+  it should "decline transactions not meeting min fee" in {
+    val (us, bh) = createUtxoState()
+    val genesis = validFullBlock(None, us, bh, Random)
+    val wus = WrappedUtxoState(us, bh, stateConstants).applyModifier(genesis).get
+    val txs = validTransactionsFromUtxoState(wus, Random)
+    val maxSettings = settings.copy(nodeSettings = settings.nodeSettings.copy(minimalFeeAmount = Long.MaxValue))
+    val pool = ErgoMemPool.empty(maxSettings)
+    txs.foreach { tx =>
+      pool.process(tx, us)._2.isInstanceOf[ProcessingOutcome.Declined] shouldBe true
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala]: Check that the reason to be Declined is that fee is small

File: src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala [hunk 3]
```diff
@@ -37,24 +37,37 @@ class ErgoMemPoolSpec extends FlatSpec
...
+        pool.process(tx, us)._2.isInstanceOf[ProcessingOutcome.Declined] shouldBe true)
+    }
+  }
+
+  it should "decline transactions not meeting min fee" in {
+    val (us, bh) = createUtxoState()
+    val genesis = validFullBlock(None, us, bh, Random)
+    val wus = WrappedUtxoState(us, bh, stateConstants).applyModifier(genesis).get
+    val txs = validTransactionsFromUtxoState(wus, Random)
+    val maxSettings = settings.copy(nodeSettings = settings.nodeSettings.copy(minimalFeeAmount = Long.MaxValue))
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala]: Also check that the same transaction will be accepted if `minimalFeeAmount = 0`

Comment (oskin1) [reply src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala]: Done

File: src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala (hunk 2 above)
Comment (oskin1) [reply src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala]: Done

**Pull request merged**